src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/RegionDropdown.tsx

```diff
@@ -0,0 +1,67 @@
+import React, {FC, useCallback, useContext, useMemo} from 'react'
+import {ComponentStatus, Dropdown} from '@influxdata/clockface'
+
+// Components
+import {CreateOrgContext} from 'src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrganizationContext'
+import {Cluster} from 'src/client/unityRoutes'
+
+const SINGLE_PROVIDER_WIDTH = 375
+
+export const RegionDropdown: FC = () => {
+  const {clusters, currentProvider, currentRegion, changeCurrentRegion} =
+    useContext(CreateOrgContext)
+  const handleRegionChange = useCallback(
+    region => {
+      changeCurrentRegion(region)
+    },
+    [changeCurrentRegion]
+  )
+  const regions = useMemo(() => {
+    return clusters?.[currentProvider]
+  }, [clusters, currentProvider])
```

Comment:
For most cases, this is a simple `O(1)` lookup on hashmap. Memoizing it will probably make it less performant, since the memoized lookup now has to do a shallow check of `clusters` to see if it's changed before using the memoized function. So it checks the entire object and compares keys instead of just accessing a single key. I think the `useMemo` should probably get axed here

Reply:
you are absolutely correct and will remove it.